src/irr/asset/material_compiler/CMaterialCompilerGLSLBackendCommon.cpp

```diff
@@ -49,6 +49,13 @@ namespace instr_stream
 			dst = core::bitfieldInsert(dst, parent>>BITFIELDS_SHIFT_MASKFLAG, BITFIELDS_SHIFT_MASKFLAG, 1);
 		}
 
+		// Extra operations performed on instruction just before it is pushed on stack
+		virtual void onBeforeStackPush(instr_t& instr) const
+		{
+			if (getOpcode(instr) == OP_DIELECTRIC && isTwosided(instr))
+				setOpcode(instr, OP_THINDIELECTRIC);
```

Comment:
if roughdielectric then assert or error emit

Reply:
TODO

Reply:
still TODO

Reply:
not needed any more since twosided BSDF is illegal anyway

Reply:
+ we wont even have twosided flag